### PR TITLE
Upgrade Elasticsearch from 1.7 to 5.x

### PR DESCRIPTION
**Pull request by viclim**
### Notable Changes
- Upgraded to Elasticsearch 5.6.5.
- Elasticsearch 6 **not** supported, due to `groovy` being dropped from v6.

### CODE REVIEW

**Comment by viclim** [comment]
I will remove the additional comments once this is reviewed and approved.

The `sleep 1` statements added in the spec to workaround the version conflict error `Elasticsearch::Transport::Transport::Errors::Conflict`, caused by the different `delete_by_query` behaviour in ES5.

**Comment by viclim** [comment]
`number_of_shards` is now configured within rails. 
It's set to `1` in `/config/elasticsearch.yml`

File: spec/models/album_detection_photo_iterator_spec.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 describe AlbumDetectionPhotoIterator, "run" do
   before do
     FlickrPhoto.delete_all
+    sleep 1
```
Comment (nickmarden) [inline spec/models/album_detection_photo_iterator_spec.rb]: I thought that @noremmie had suggested a different way to get around this by setting an explicit `on_conflict` handler or something?

File: README.md [hunk 2]
```diff
@@ -33,14 +33,22 @@ We use bundler to manage gems. You can install bundler and other required gems l
...
-You can generally leave the defaults in elasticsearch.yml as they are, but there are two exceptions:
+You can generally leave the defaults in elasticsearch.yml as they are, with the exception of:
+
+#### Scripting
+
+[Scripting](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/modules-scripting-security.html#security-script-source) is disabled by default in Elasticsearch 5.6. Set both `script.inline` and `script.stored` to `true`:
+
+    script:
+      inline: true
+      stored: true
```
Comment (nickmarden) [inline README.md]: I think this is incomplete. I think that either the user needs to disable X-Pack security, or they need to configure a password in the Rails app's `elasticsearch.yml` file.

Comment (nickmarden) [reply README.md]: My mistake! I somehow missed the new `config/elasticsearch.yml` in this PR :-/

**Comment by nickmarden** [comment]
Right now this is blocked on waiting to hear why @noremmie's solution for `on_conflict` is less preferable that a weird `sleep` statement in our specs :-)

**Comment by viclim** [comment]
I missed the [comment](https://www.elastic.co/guide/en/elasticsearch/reference/5.4/docs-delete-by-query.html#_url_parameters) from @noremmie in the Slack channel. Am looking into it.

**Comment by viclim** [comment]
Removed the `sleep 1` from the specs.
Using `conflicts: :proceed` options when doing `delete_by_query`.

**Pull request merged**